lib/stat_tracker.rb

```diff
@@ -41,202 +41,211 @@ def count_of_teams
-      coaches_with_games_won_array = games_won_into_hash_helper.keys
-      coaches_with_games_won_and_played_array = coaches_with_games_played_array & coaches_with_games_won_array
-      coaches_with_games_won_and_played_array
+    # Returns name of coach with greatest win percentage (games won divided by total of games played)
+    season_games.max_by do |coach, games|
+      games[0].to_f / games.sum
+    end.first
   end
 
-  def coaches_winning_percentage_hash_helper
-    coaches_winning_percentage_hash = {}
-    coaches_with_games_played_and_won_array_helper.each do |coach|
+  def worst_coach(season_id)
+    # Find all games in season & return array of those games' ids
+    season_games = @games.find_all do |game|
+      game["season"] == season_id
+    end.map {|game| game["game_id"]}
 
-      coaches_winning_percentage_hash[coach] = ((games_won_into_hash_helper[coach] / total_games_played_by_coach_helper[coach] * 100))
+    # Find all those games in @game_teams & return array of CSV row objects
+    season_games = @game_teams.find_all do |game|
+      season_games.include?(game["game_id"])
     end
-    coaches_winning_percentage_hash
+
+    # Creates hash with 'head_coach' as key and value as an array with first
+    # element as integer of games won and second element integer of all other games (losses or ties)
+    season_games = season_games.group_by do |game|
+      game["head_coach"]
+    end.transform_values! {|games| games.partition {|game| game["result"] == "WIN"}.map(&:length)}
+
+    # Returns name of coach with greatest win percentage (games won divided by total of games played)
+    season_games.min_by do |coach, games|
+      games[0].to_f / games.sum
+    end.first
   end
 
-  def winningest_coach
-    name_of_coach_with_number_variable = coaches_winning_percentage_hash_helper.find_all do |key, value|
-      value == coaches_winning_percentage_hash_helper.values.max
-    end
-    name_of_coach_array_variable = []
-  name_of_coach_with_number_variable.each do |name_and_number|
-  name_of_coach_array_variable << name_and_number[0]
-    end
-    name_of_coach_array_variable.join(", ")
+  def most_accurate_team(season_id)
+    # Find all games in season & return array of those games' ids"
+  season_games = @games.find_all do |game|
+    game["season"] == season_id
+  end.map {|game| game["game_id"]}
+
+  # Find all those games in @game_teams & return array of CSV row objects
+  season_games = @game_teams.find_all do |game|
+    season_games.include?(game["game_id"])
   end
 
-  def worst_coach_if_someone_doesnt_have_any_wins_helper
-    worst_coach_array = []
-    total_games_played_by_coach_helper.keys.each do |coach|
-      if !games_won_into_hash_helper.keys.include?(coach)
-        worst_coach_array << coach
+  # Creates hash with 'team_id' as key and value as total shots taken for the season
+  total_shots_by_team = {}
+    season_games.each do |game|
+    if total_shots_by_team.include?(game["team_id"])
```

Comment:
Nice use of to_i here to convert values to Integers and add those values together